kuka_rox_hardware_interface/src/robot_manager_node.cpp

```diff
@@ -309,21 +319,112 @@ RobotManagerNode::on_deactivate(const rclcpp_lifecycle::State &)
     RCLCPP_ERROR(get_logger(), "Could not stop controllers");
     return ERROR;
   }
+  controller_handler_.ApproveControllerDeactivation();
 
   RCLCPP_INFO(get_logger(), "Successfully stopped controllers");
   return SUCCESS;
 }
 
 bool RobotManagerNode::onControlModeChangeRequest(int control_mode)
 {
-  if (control_mode_map_.find(control_mode) != control_mode_map_.end()) {
+  try {
+    RCLCPP_INFO(get_logger(), "Control mode change process has started");
+    if (control_mode == kroshu_ros2_core::ControllerHandler::CARTESIAN_POSITION_CONTROL_MODE ||
```

Comment:
what if I'm changing to the control mode with the enum value 456? I think we should also handle that case

Reply:
The controller handler catch that